infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -708,6 +728,33 @@ public void updateTopology(LogReplicationClusterInfo.TopologyConfigurationMsg to
         return null;
     }
 
+    @Override
+    public boolean forceSnapshotSync() {
+        if (localClusterDescriptor.getRole() == ClusterRole.STANDBY) {
+            log.error("This forceSnapshotSync command is not supported on standby cluster.");
+            return false;
+        }
+
+        DiscoveryServiceEvent event = new DiscoveryServiceEvent(DiscoveryServiceEventType.ENFORCE_SNAPSHOT_SYNC, UUID.randomUUID());
+        input(new DiscoveryServiceEvent(DiscoveryServiceEventType.ENFORCE_SNAPSHOT_SYNC, UUID.randomUUID()));
+
+        // Block until the full snapshot sync has been started on all the standby clusters.
```

Comment:
I think we can not guarantee that all sync have been started... We can only guarantee that all sync start event have been added to the queue...